c7n/resources/dynamodb.py

```diff
@@ -120,6 +122,48 @@ def __call__(self, r):
         return super().__call__(r.get(self.annotation_key, {}))
 
 
+@Table.filter_registry.register('cross-account')
+class CrossAccountTable(CrossAccountAccessFilter):
+
+    permissions = ('dynamodb:GetResourcePolicy',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('dynamodb')
+        for r in resources:
```

Comment:
Might make sense to looks for policy first before making an API call
````
```
if 'c7n:Policy' in r:
   continue
```
````